lib/ansible/modules/user.py

```diff
@@ -2774,7 +2774,21 @@ def create_user(self):
         if rc is not None and rc != 0:
             self.module.fail_json(name=self.name, msg=err, rc=rc)
 
-        if self.password is not None:
+        # Since -D is given to adduser above, the user is created locked by
+        # default, so cannot be accessed even if they have a valid SSH key
+        # in place. If a password is given, we `chpasswd` and the account
+        # unlocks. But otherwise, we have to unlock it manually.
+        #
+        # https://github.com/ansible/ansible/issues/68676
+        # https://bugs.busybox.net/show_bug.cgi?id=10981
+        if self.password is None:
+            cmd = [self.module.get_bin_path('passwd', True)]
+            cmd += ['-u', self.name]
```

Comment:
I think the problem with doing this on an account with no password set is that it will result in an account with _no_ password. So blindly unlocking the account is probably not a good idea.

What we probably need to do is set the password to `*` if one is not provided since that effectively creates an unlocked account with no password, which is what is expected.

Reply:
```
bash-5.0# adduser -D testuser
bash-5.0# grep testuser /etc/shadow
testuser:!:18470:0:99999:7:::
bash-5.0# passwd -u testuser
passwd: password for testuser changed by root
bash-5.0# grep testuser /etc/shadow
testuser::18470:0:99999:7:::
```

That looks to me like an account with no password.

Reply:
New approach edits /etc/shadow